playground/backend/internal/setuper/setuper.go

```diff
@@ -0,0 +1,91 @@
+package setuper
+
+import (
+	pb "beam.apache.org/playground/backend/internal/api/v1"
+	"beam.apache.org/playground/backend/internal/environment"
+	"beam.apache.org/playground/backend/internal/executors"
+	"beam.apache.org/playground/backend/internal/fs_tool"
+	"beam.apache.org/playground/backend/internal/logger"
+	"beam.apache.org/playground/backend/internal/validators"
+	"github.com/google/uuid"
+)
+
+// SetupLifeCycle creates fs_tool.LifeCycle and prepares files and folders needed to code processing
+func SetupLifeCycle(sdk pb.Sdk, code string, pipelineId uuid.UUID, workingDir string) (*fs_tool.LifeCycle, error) {
+	// create file system service
+	lc, err := fs_tool.NewLifeCycle(sdk, pipelineId, workingDir)
+	if err != nil {
+		logger.Errorf("%s: RunCode(): NewLifeCycle(): %s\n", pipelineId, err.Error())
+		return nil, err
+	}
+
+	// create folders
+	err = lc.CreateFolders()
+	if err != nil {
+		logger.Errorf("%s: RunCode(): CreateFolders(): %s\n", pipelineId, err.Error())
+		return nil, err
+	}
+
+	// create file with code
+	_, err = lc.CreateExecutableFile(code)
+	if err != nil {
+		logger.Errorf("%s: RunCode(): CreateExecutableFile(): %s\n", pipelineId, err.Error())
+		return nil, err
+	}
+	return lc, nil
+}
+
+// SetupCompileBuilder returns executors.CompileBuilder with validators and compiler based on sdk
+func SetupCompileBuilder(lc *fs_tool.LifeCycle, sdk pb.Sdk, executorConfig *environment.ExecutorConfig) *executors.CompileBuilder {
+	filePath := lc.GetAbsoluteExecutableFilePath()
+	val := SetupValidators(sdk, filePath)
+
+	compileBuilder := executors.NewExecutorBuilder().
+		WithValidator().
+		WithSdkValidators(val).
+		WithCompiler()
+
+	switch sdk {
+	case pb.Sdk_SDK_JAVA:
```

Comment:
Shouldn't we return an error on the wrong SDK?

Reply:
We should, because if someone passes here the wrong SDK, this function will return the lousy builder, but no one will know about that.

Reply:
Changed.